datastreams/windower/tumbling.go

```diff
@@ -0,0 +1,104 @@
+package windower
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/elastiflow/pipelines/datastreams/internal/partition"
+	"github.com/elastiflow/pipelines/datastreams/internal/pipes"
+)
+
+type status struct {
+	started bool
+	mu      sync.Mutex
+}
+
+func newStatus() *status {
+	return &status{
+		started: false,
+	}
+}
+
+func (t *status) running() bool {
+	t.mu.Lock()
+	defer t.mu.Unlock()
+	return t.started
+}
+
+func (t *status) set(state bool) {
+	t.mu.Lock()
+	defer t.mu.Unlock()
+	t.started = state
+}
```

Comment:
I'd recommend using a `atomic.Bool` here:

```suggestion
import (
	"context"
	"sync"
	"sync/atomic"
	"time"

	"github.com/elastiflow/pipelines/datastreams/internal/partition"
	"github.com/elastiflow/pipelines/datastreams/internal/pipes"
)

type status struct {
	started atomic.Bool
}

func (s *status) running() bool {
	return s.started.Load()
}

func (s *status) SetRunning(v bool) {
	s.started.Store(v)
}
```